### ISSUE TITLE
incorrect number of tokens in target selection

### ISSUE THREAD

**Issue by r00t-3xp10it** [r00t-3xp10it/morpheus#1]

**Comment by spiritedwolf** [comment]
yeah it is the same error which i was talking about.

**Comment by spiritedwolf** [comment]
So means we can't patch this bug?

**Comment by r00t-3xp10it** [comment]
<br />
this is not a bug, its a morpheus (ettercap) missconfiguration...

    If your pc its using IPv4 only and morpheus its configurated to use IPv6
    "then morpheus will produce the above display (error)"

    if your pc its using IPv6 and morpheus its configurated to use IPv4
    "then morpheus will produce the above display (error)"

<br />

**Comment by Ekultek** [comment]
You can patch this by decoding the IPv6 it won't always work, buts it's a temporary work around

**Comment by r00t-3xp10it** [comment]
hey Ekultek explain that better (iam curiouse)...

**Comment by Ekultek** [comment]
@r00t-3xp10it you can attempt to decode an IPv6 to IPv4 sometimes, it doesn't always work because the difference in bits but every now and then you'll get lucky. So you can attempt to decode it, and catch the error if it doesn't work, if it does then there ya go, IPv4 address. Read this http://www.nish.com/2009/convert-ipv4-into-ipv6/

**Comment by r00t-3xp10it** [comment]
i understand what are you talking about now.
but this '**issue**' its only a miss-configuration in morpheus '**settings**' file...
"Thats no need to decode from IPv6 to IPv4, the user only needs to  config
the proper value in '**setting**' file to be used befor runnig the tool"

<br /><br />

> settings file will '**activate**' ettercap ipv6 support or '**disable**' the ipv6 and use ipv4 ...

ettercap ipv4 example
`ettercap -T -Q -i wlan0 -F morpheus/output/img_replace.ef -M ARP /$target/ /$GaTe/`

ettercap ipv6 example
`ettercap -T -Q -i wlan0 -F morpheus/output/img_replace.ef -M ARP /$target// /$GaTe//`


But that link that you have shared its going to be ussefull, thanks...

**Comment by Ekultek** [comment]
Awh, I got ya. Well on another note, you could auto config the file from the script? Run an initialization of it and put in the correct configuration for the file during the initialization

**Comment by r00t-3xp10it** [comment]
yes but i dont know how to write the funtion that allow me to check if user its using IPv4 or IPv6
"to proper auto.config settings", thats the reason why the existence of 'settings' file...
